### PR TITLE
mvcc: commit may fail unexpectedly after meta splits.

### PR DESCRIPTION
**Pull request by disksing**
@ngaut @siddontang

### CODE REVIEW

**Comment by siddontang** [comment]
LGTM

File: src/storage/mvcc/txn.rs [hunk 1]
```diff
@@ -271,6 +271,36 @@ impl<'a> MvccSnapshot<'a> {
...
+                         -> Result<Option<u64>> {
+        if let Some(x) = first_meta.iter_items().find(|x| x.get_start_ts() <= start_ts) {
+            return if x.get_start_ts() == start_ts {
+                Ok(Some(x.get_commit_ts()))
+            } else {
+                Ok(None)
+            };
+        }
+        let mut next = first_meta.next_index();
+        loop {
```
Comment (BusyJay) [inline src/storage/mvcc/txn.rs]: Use while let instead.

**Comment by BusyJay** [comment]
rest LGTM

**Comment by BusyJay** [comment]
LGTM

**Pull request merged**